src/App.tsx

```diff
@@ -1,61 +1,127 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId)
+      || null;
+}
+
+export const initialTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const addTodo = (newTodo: Todo) => {
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+
+    setHasTitleError(false);
+    setHasUserError(false);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const maxId = Math.max(...todos.map(todo => todo.id));
+
+    addTodo({
+      id: maxId + 1,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
+    });
+
+    reset();
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
```

Comment:
The 'action' and 'method' attributes are not needed in the form element because the form submission is being handled manually with the `onSubmit` event.